badgecheck/state.py

```diff
@@ -5,5 +5,17 @@
 }
 
 
+# Tasks
 def filter_active_tasks(state):
     return [t for t in state.get('tasks') if not t.get('complete')]
+
+
+# Graph
+def get_node_by_id(state, node_id):
+    """
+    Filter state to return first node that matches the requested id.
+    :param state: state object with "graph" property as a list
+    :param node_id: IRI-format string
```

Comment:
If caller is looking for a blank node, they'll have to pass node_id in the form `'_b:{}'.format(node_id)` then, correct? Or once a node has been blank-node-ized, we consider the entire IRI to be the (new) node ID?